spaceship/lib/spaceship/tunes/build_train.rb

```diff
@@ -118,11 +118,11 @@ def latest_build
 
       # @param (testing_type) internal or external
       def update_testing_status!(new_value, testing_type, build = nil)
-        data = client.build_trains(self.application.apple_id, testing_type, platform: self.application.platform)
-
         build ||= latest_build if testing_type == 'external'
         testing_key = "#{testing_type}Testing"
 
+        data = client.build_trains(self.application.apple_id, testing_type, platform: build.platform)
```

Comment:
Using `self.application.platform` doesn't make any sense, because the application has binaries for both platforms. Instead we use `build.platform` so that build trains for the correct platform is updated.